chef/cookbooks/ohai/files/default/plugins/linux/block_device_extended.rb

```diff
@@ -31,11 +31,11 @@
     parts = File.realpath(dir, "/sys/block/").split("/")
     # example:
     # /sys/devices/pci0000:00/0000:00:02.0/0000:03:00.0/host0/target0:1:4/0:1:4:0/block/sdb
-    parts = parts[0..4] + ["driver"]
```

Comment:
There can be a driver entry in both locations:
~~~
# ll /sys/devices/pci0000:00/0000:00:02.2/
driver -> ../../../bus/pci/drivers/pcieport

# ll /sys/devices/pci0000:00/0000:00:02.2/0000:02:00.0
driver -> ../../../../bus/pci/drivers/hpsa
~~~
The driver at the deeper level seems more appropriate though.